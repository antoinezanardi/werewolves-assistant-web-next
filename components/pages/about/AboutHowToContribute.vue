<template>
  <div id="about-how-to-contribute">
    <h2
      id="about-how-to-contribute-title"
      class="flex items-center"
    >
      <i class="fa fa-2x fa-handshake-angle me-3 text-green-500"/>

      <span>
        {{ $t('components.AboutHowToContribute.howToContribute') }}
      </span>
    </h2>

    <VuePrimeDivider/>

    <p id="about-how-to-contribute-first-section">
      {{ $t("components.AboutHowToContribute.thereAreManyWaysToContribute") }}
    </p>

    <ul id="about-how-to-contribute-second-section">
      <li>
        {{ $t("components.AboutHowToContribute.openSourceProject") }}
        <div class="flex items-center justify-center my-3">
          <GitHubRepositoryButton
            id="about-how-to-contribute-github-button"
            :text-button="$t('components.AboutHowToContribute.accessGitHubRepository')"
          />
        </div>
      </li>

      <li>
        {{ $t("components.AboutHowToContribute.thisProjectIsFree") }}

        <div class="flex items-center justify-center my-3">
          <BuyMeACoffeeButton/>
        </div>
      </li>

      <li>
        {{ $t("components.AboutHowToContribute.talkAboutThisProject") }}
      </li>
    </ul>
  </div>
</template>

<script setup lang="ts">
import BuyMeACoffeeButton from "~/components/shared/external/BuyMeACoffeeButton/BuyMeACoffeeButton.vue";
import GitHubRepositoryButton from "~/components/shared/external/GitHubRepositoryButton/GitHubRepositoryButton.vue";
</script>