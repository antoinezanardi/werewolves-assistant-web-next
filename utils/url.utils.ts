function removeTrailingSlashes(url: string): string {
  return url.replace(/\/+$/u, "");
}

export { removeTrailingSlashes };