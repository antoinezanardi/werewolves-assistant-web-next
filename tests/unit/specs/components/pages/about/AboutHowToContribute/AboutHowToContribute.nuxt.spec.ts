import type { mount } from "@vue/test-utils";

import AboutHowToContribute from "~/components/pages/about/AboutHowToContribute.vue";
import type GitHubRepositoryButton from "~/components/shared/external/GitHubRepositoryButton/GitHubRepositoryButton.vue";
import { mountSuspendedComponent } from "~/tests/unit/utils/mount.utils";

describe("About How To Contribute Component", () => {
  let wrapper: ReturnType<typeof mount<typeof AboutHowToContribute>>;

  beforeEach(async() => {
    wrapper = await mountSuspendedComponent(AboutHowToContribute);
  });

  it("should render component and match snapshot when mounted.", () => {
    expect(wrapper).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe("Title", () => {
    it("should translate title when rendered.", () => {
      const title = wrapper.find<HTMLHeadElement>("#about-how-to-contribute-title");

      expect(title.text()).toBe("How to contribute to this project?");
    });
  });

  describe("Sections", () => {
    it("should translate first section when rendered.", () => {
      const firstSection = wrapper.find<HTMLParagraphElement>("#about-how-to-contribute-first-section");

      expect(firstSection.text()).toBe("There are several ways to contribute to this project:");
    });

    it("should translate second section for each bullet item when rendered.", () => {
      const secondSection = wrapper.find<HTMLUListElement>("#about-how-to-contribute-second-section");
      const secondSectionBulletItems = secondSection.findAll<HTMLLIElement>("li");

      expect(secondSectionBulletItems).toHaveLength(3);
      expect(secondSectionBulletItems[0].text()).toBe("The Werewolf Assistant is an open-source project! Available on GitHub, whether you are a developer, graphic designer, " +
      "or even a tester, your help is welcome. If you like the project, feel free to star it or follow the creator!");
      expect(secondSectionBulletItems[1].text()).toBe("This project is intended to remain a free service. However, both hosting and tool infrastructure are paid for. " +
      "If you would like to contribute or appreciate the creator's work, you can make a donation on their Buy Me A Coffee page.");
      expect(secondSectionBulletItems[2].text()).toBe("Talk about this project around you! " +
      "Word of mouth will encourage many role-play enthusiasts to embark on the grand adventure of the game master with the Werewolf Assistant.");
    });

    it("should translate gitHub text button when rendered.", () => {
      const gitHubButton = wrapper.findComponent<typeof GitHubRepositoryButton>("#about-how-to-contribute-github-button");

      expect(gitHubButton.props("textButton")).toBe("Access the project on GitHub");
    });
  });
});