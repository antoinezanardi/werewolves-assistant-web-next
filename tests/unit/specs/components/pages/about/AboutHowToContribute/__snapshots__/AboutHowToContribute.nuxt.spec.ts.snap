// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`About How To Contribute Component > should render component and match snapshot when mounted. 1`] = `
"<div id="about-how-to-contribute">
  <h2 id="about-how-to-contribute-title" class="align-items-center d-flex"><i class="fa fa-2x fa-handshake-angle me-3 text-success"></i><span>How to contribute to this project?</span></h2>
  <hr>
  <p id="about-how-to-contribute-first-section">There are several ways to contribute to this project:</p>
  <ul id="about-how-to-contribute-second-section">
    <li>The Werewolf Assistant is an open-source project! Available on GitHub, whether you are a developer, graphic designer, or even a tester, your help is welcome. If you like the project, feel free to star it or follow the creator! <div class="align-items-center d-flex justify-content-center my-3">
        <git-hub-repository-button-stub textbutton="Access the project on GitHub" id="about-how-to-contribute-github-button"></git-hub-repository-button-stub>
      </div>
    </li>
    <li>This project is intended to remain a free service. However, both hosting and tool infrastructure are paid for. If you would like to contribute or appreciate the creator's work, you can make a donation on their Buy Me A Coffee page. <div class="align-items-center d-flex justify-content-center my-3">
        <buy-me-a-coffee-button-stub></buy-me-a-coffee-button-stub>
      </div>
    </li>
    <li>Talk about this project around you! Word of mouth will encourage many role-play enthusiasts to embark on the grand adventure of the game master with the Werewolf Assistant.</li>
  </ul>
</div>"
`;
